concurrency-limits-core/src/main/java/com/netflix/concurrency/limits/strategy/SimpleStrategy.java

```diff
@@ -23,21 +27,32 @@ public SimpleStrategy() {
     }
     
     public SimpleStrategy(MetricRegistry registry) {
+        this(registry, () -> {});
+    }
+
+    // Test Only
+    SimpleStrategy(MetricRegistry registry, Runnable acquireBarrier) {
         this.inflightMetric = registry.registerDistribution(MetricIds.INFLIGHT_GUAGE_NAME);
         registry.registerGauge(MetricIds.LIMIT_GUAGE_NAME, this::getLimit);
+        this.acquireBarrier = acquireBarrier;
     }
     
     @Override
     public Token tryAcquire(T context) {
-        final int currentBusy = busy.get();
-        if (currentBusy >= limit) {
-            inflightMetric.addSample(currentBusy);
-            return Token.newNotAcquired(currentBusy);
+        // Spin in a CAS loop to guard against modifications after reading.
+        while (true) {
```

Comment:
I'm not super familiar with the code, but unless consumers are expected to .release() quickly, it feels risky dangerous to hook callers in the CAS loop. Would it be safer to simply:

```
        final int inflight = currentBusy + 1;
        if (!busy.compareAndSet(currentBusy, inflight)) {
            return Token.newNotAcquired(busy.get());
        }
```

At L36 of the base revision? Callers can still retry if they wish.

This has a caveat: the value returned in newNotAcquired is not guaranteed to be precise.